test-data/unit/fine-grained-cycles.test

```diff
@@ -205,8 +205,7 @@ def h() -> None:
 [out]
 ==
 a.py:3: error: Invalid type "b.C"
-b.py:6: error: Need type annotation for 'c'
-b.py:7: error: Cannot determine type of 'c'
+b.py:7: error: C? has no attribute "g"
```

Comment:
Hmm do we now propagate unbound types in type inference? Not sure if that's a good or bad thing.

Reply:
> Not sure if that's a good or bad thing.

This is rather bad, but I believe the right solution is https://github.com/python/mypy/issues/4987.